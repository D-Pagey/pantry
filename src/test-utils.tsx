import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import React, { FC, ReactNode } from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { FirebaseContext } from './components/ProviderFirebase';

type AllTheProvidersProps = {
    children: ReactNode;
    customContext?: any;
};

const AllTheProviders: FC<AllTheProvidersProps> = ({ children, customContext }) => (
    <FirebaseContext.Provider
        value={{
            categories: [],
            deleteFoodItem: () => () => null,
            expiringCount: 0,
            isAuthed: false,
            isCheckingAuth: false,
            fridge: [],
            signOut: () => null,
            updateFridge: () => null,
            user: {
                email: null,
                name: null
            },
            ...customContext
        }}
    >
        <MemoryRouter>{children}</MemoryRouter>
    </FirebaseContext.Provider>
);

const customRender = (ui: any, customContext?: any, options?: any) =>
    render(ui, { wrapper: () => <AllTheProviders customContext={customContext}>{ui}</AllTheProviders>, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
