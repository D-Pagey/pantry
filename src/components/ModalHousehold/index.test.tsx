import React from 'react';
import { render } from '../../test-utils';
import { ModalHousehold } from '.';

const props = {
    isAdmin: false,
    onModalClose: () => null
};

describe('ModalHousehold component', () => {
    it('should render', () => {
        const { container } = render(<ModalHousehold {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
