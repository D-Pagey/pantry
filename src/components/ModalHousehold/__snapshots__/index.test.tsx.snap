// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHousehold component should render 1`] = `
.c2 {
  background-color: #348841;
  border: 0;
  border-radius: 5px;
  color: white;
  cursor: pointer;
  padding: 1rem;
  min-width: 68px;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.c1 {
  font-size: 28px;
  font-weight: 400;
  margin: 0 0 20px;
}

.c3 {
  background-color: white;
  border: 1px solid rgb(66,133,244);
  color: rgb(66,133,244);
  font-size: 1.25rem;
  margin: 0 0 20px;
  width: 200px;
}

.c4 {
  background-color: white;
  border: 1px solid #348841;
  color: #348841;
  font-size: 1.25rem;
  margin: 0 0 20px;
  width: 200px;
}

.c5 {
  background-color: white;
  border: 1px solid #EB5757;
  color: #EB5757;
  font-size: 1.25rem;
  margin: 0 0 20px;
  width: 200px;
}

<div
  class="c0"
>
  <h3
    class="c1"
  >
    Household Options:
  </h3>
  <button
    class="c2 c3"
    color="rgb(66, 133, 244)"
    type="button"
  >
    Cancel Invite
  </button>
  <button
    class="c2 c4"
    color="#348841"
    type="button"
  >
    Promote to Owner
  </button>
  <button
    class="c2 c5"
    color="#EB5757"
    type="button"
  >
    Remove User
  </button>
  <button
    class="c2 c5"
    color="#EB5757"
    type="button"
  >
    Leave household
  </button>
  <button
    class="c2"
    type="button"
  >
    Close
  </button>
</div>
`;
